From adfa8ab8dc74734ac8bc148d21b266b9c3087946 Mon Sep 17 00:00:00 2001
From: Prince <mnisiddique@ael-bd.com>
Date: Mon, 8 Jul 2024 02:08:40 +0600
Subject: [PATCH] build: migrates style of applying flutter gradle plugin from
 imperative to declarative. desc: reason for doing this is following warning
 'You are applying Flutter's main Gradle plugin imperatively using the apply
 script method, which is deprecated and will be removed in a future release.
 Migrate to applying Gradle plugins with the declarative plugins block:
 https://flutter.dev/go/flutter-gradle-plugin-apply.'

---
 android/app/build.gradle | 30 +++++++++++-------------------
 android/build.gradle     | 20 --------------------
 android/settings.gradle  | 32 ++++++++++++++++++++++----------
 3 files changed, 33 insertions(+), 49 deletions(-)

diff --git a/android/app/build.gradle b/android/app/build.gradle
index 445a29f9..2a42a9c7 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -1,3 +1,14 @@
+plugins {
+    id "com.android.application"
+// START: FlutterFire Configuration
+    id "com.google.firebase.crashlytics"
+    id "com.google.gms.google-services"
+// END: FlutterFire Configuration
+    id "kotlin-android"
+    id "dev.flutter.flutter-gradle-plugin"
+}
+
+
 def localProperties = new Properties()
 def localPropertiesFile = rootProject.file('local.properties')
 if (localPropertiesFile.exists()) {
@@ -6,18 +17,6 @@ if (localPropertiesFile.exists()) {
     }
 }
 
-
-
-def flutterRoot = localProperties.getProperty('flutter.sdk')
-if (flutterRoot == null) {
-    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
-}
-
-//For Playstore build Update the flutter versionCode and  flutterVersionName in app\build.gradle file
-//i.e., flutterVersionCode = '4032' to flutterVersionCode = ‘4033’
-//flutterVersionName = '1.0.3' to  flutterVersionName=’1.0.4’’
-
-
 def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
 if (flutterVersionCode == null) {
     flutterVersionCode = '1'
@@ -28,13 +27,6 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0.0'
 }
 
-apply plugin: 'com.android.application'
-// START: FlutterFire Configuration
-apply plugin: 'com.google.gms.google-services'
-apply plugin: 'com.google.firebase.crashlytics'
-// END: FlutterFire Configuration
-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
-
 def keystoreProperties = new Properties()
 def keystorePropertiesFile = rootProject.file('key.properties')
 if (keystorePropertiesFile.exists()) {
diff --git a/android/build.gradle b/android/build.gradle
index dc01f5f8..f3604120 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,27 +1,7 @@
-buildscript {
-    ext.kotlin_version = '1.8.21'
-    repositories {
-        google()
-        mavenCentral()
-//        jcenter()
-    }
-
-    dependencies {
-        classpath 'com.android.tools.build:gradle:7.3.1'
-        // START: FlutterFire Configuration
-        classpath 'com.google.gms:google-services:4.3.15'
-        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.8.1'
-        // END: FlutterFire Configuration
-        // classpath 'com.android.tools.build:gradle:4.1.0'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-    }
-}
-
 allprojects {
     repositories {
         google()
         mavenCentral()
-//        jcenter()
     }
 }
 
diff --git a/android/settings.gradle b/android/settings.gradle
index 5a2f14fb..1ed1525b 100644
--- a/android/settings.gradle
+++ b/android/settings.gradle
@@ -1,15 +1,27 @@
-include ':app'
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
+        file("local.properties").withInputStream { properties.load(it) }
+        def flutterSdkPath = properties.getProperty("flutter.sdk")
+        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
+        return flutterSdkPath
+    }()
 
-def flutterProjectRoot = rootProject.projectDir.parentFile.toPath()
+    includeBuild("$flutterSdkPath/packages/flutter_tools/gradle")
 
-def plugins = new Properties()
-def pluginsFile = new File(flutterProjectRoot.toFile(), '.flutter-plugins')
-if (pluginsFile.exists()) {
-    pluginsFile.withReader('UTF-8') { reader -> plugins.load(reader) }
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
 }
 
-plugins.each { name, path ->
-    def pluginDirectory = flutterProjectRoot.resolve(path).resolve('android').toFile()
-    include ":$name"
-    project(":$name").projectDir = pluginDirectory
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "7.3.1" apply false
+    id "org.jetbrains.kotlin.android" version "1.8.21" apply false
+    id "com.google.gms.google-services" version "4.3.15" apply false
+    id "com.google.firebase.crashlytics" version "2.8.1" apply false
 }
+
+include ":app"
\ No newline at end of file
-- 
2.45.1.windows.1

